src/app/modules/billing/__tests__/billing.routes.spec.ts

```diff
@@ -0,0 +1,244 @@
+import { flatten, sortBy, times } from 'lodash'
+import mongoose from 'mongoose'
+import { errAsync } from 'neverthrow'
+import supertest, { Session } from 'supertest-session'
+
+import getFormModel from 'src/app/models/form.server.model'
+import getLoginModel from 'src/app/models/login.server.model'
+import { AuthType, IUserSchema, ResponseMode } from 'src/types'
+
+import { createAuthedSession } from 'tests/integration/helpers/express-auth'
+import { setupApp } from 'tests/integration/helpers/express-setup'
+import { buildCelebrateError } from 'tests/unit/backend/helpers/celebrate'
+import dbHandler from 'tests/unit/backend/helpers/jest-db'
+
+import { DatabaseError } from '../../core/core.errors'
+import { BillingFactory } from '../billing.factory'
+import { BillingRouter } from '../billing.routes'
+
+const app = setupApp('/billing', BillingRouter, {
+  setupWithAuth: true,
+})
+
+const FormModel = getFormModel(mongoose)
+const LoginModel = getLoginModel(mongoose)
+
+describe('billing.routes', () => {
+  let request: Session
+  let defaultUser: IUserSchema
+
+  beforeAll(async () => await dbHandler.connect())
+  beforeEach(async () => {
+    request = supertest(app)
+    const { user } = await dbHandler.insertFormCollectionReqs()
+    defaultUser = user
+  })
+  afterEach(async () => {
+    await dbHandler.clearDatabase()
+    jest.restoreAllMocks()
+  })
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('GET /billing', () => {
+    const VALID_ESRVCID_1 = 'mockEsrvcId1'
+    const VALID_ESRVCID_2 = 'mockEsrvcId2'
+    const INVALID_ESRVCID = 'invalidEsrvcId'
+    const VALID_QUERY_YR = new Date().getFullYear()
+    const VALID_QUERY_MTH = new Date().getMonth()
+
+    it('should return 200 with array of results of form SPCP logins of the given esrvcId', async () => {
+      // Arrange
+      // Log in user.
+      const session = await createAuthedSession(defaultUser.email, request)
+      // Generate login statistics.
+      // Create 3 random forms.
+      const formPromises = times(3, (idx) =>
+        FormModel.create({
+          title: `example form title ${idx}`,
```

Comment:
minor nit but maybe better to move all this data creation logic out of the test and keep the test itself readable?

Reply:
Extracted a helper method in d6dd5e0